### PR TITLE
feat: export text file filesystem plugin

### LINKED ISSUE #4244: [Task]: Implement JSON file export at Filesystem plugin
Implement JSON file create on internal cache folder, then use Sharesheet with it at the Capacitor plugin Filesystem access.
Change needs to be done on Android and iOS

### PR DESCRIPTION
**Pull request by amadeu2**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

...

## Changelog

```
- Please write a list of changes
```

## Relevant Issues

Closes #4244 
...

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by paul-boegelsack — COMMENTED** [review]
Android works like a charm 💯 🍾 

I just have trouble with iOS. Every time I try an export I get this message:

image

I tried to debug the saveTextFile method but it doesn't run into the break point, so I can't say for sure if this is a plugin problem or not. XCode and Safari log doesn't really show an error or something useful either 🤔

**Comment by begonaalvarezd** [comment]
> Android works like a charm 100 champagne
> 
> I just have trouble with iOS. Every time I try an export I get this message:
> 
> image
> 
> I tried to debug the saveTextFile method but it doesn't run into the break point, so I can't say for sure if this is a plugin problem or not. XCode and Safari log doesn't really show an error or something useful either thinking I will investigate this further.

🕵🏼 Would be nice if we could fix that notification that looks like is missing an argument to display the account alias

File: packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.m [hunk 1]
```diff
@@ -8,5 +8,5 @@
     CAP_PLUGIN_METHOD(removeProfileFolder, CAPPluginReturnPromise);
     CAP_PLUGIN_METHOD(renameProfileFolder, CAPPluginReturnPromise);
     CAP_PLUGIN_METHOD(listProfileFolders, CAPPluginReturnPromise);
-    CAP_PLUGIN_METHOD(saveRecoveryKit, CAPPluginReturnPromise);
+    CAP_PLUGIN_METHOD(saveTextFile, CAPPluginReturnPromise);
```
Comment (paul-boegelsack) [inline packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.m:11]: I'm a little further. The old method name for saveRecoveryKit was still registered so I added the new one here. But now the app crashes here (line 131):

image

@amadeu2 I sent you the error log on slack

File: packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.swift [hunk 2]
```diff
@@ -110,17 +110,29 @@ public class SecureFilesystemAccess: CAPPlugin, UIDocumentPickerDelegate {
...
+        try? data.write(to: file, options: .atomicWrite)
+        
+        let srcUrl = Capacitor.URL(fileURLWithPath: file.path, isDirectory: false)
+        var filesToShare = [Any]()
+        filesToShare.append(srcUrl)
+        let activityViewController = UIActivityViewController(activityItems: filesToShare, applicationActivities: nil)
+        
+        DispatchQueue.main.async {
+            self.bridge?.viewController?.present(activityViewController, animated: true, completion: nil)
+        }
```
Comment (paul-boegelsack) [inline packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.swift:134]: Seems the issue was that the plugin tries to present the activity controller in a background thread which seems not possible. 

https://github.com/ionic-team/capacitor/issues/841

With the DispatchQueue.main.async it works for me as intended :) 

Nice work anyway, @amadeu2  the sharing is much better than this folder selection stuff. I see no reason to block this further 👍 💯

**Review by paul-boegelsack — APPROVED** [review]

File: packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.m (hunk 1 above)
Comment (amadeu2) [reply packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.m:11]: Nice catch! 💯💪

File: packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.swift (hunk 2 above)
Comment (amadeu2) [reply packages/mobile/capacitor/plugins/secure-filesystem-access/ios/Plugin/Plugin.swift:134]: Exceltent Paul! thanks a lot for your fixes! 💪🤗

**Pull request merged**